transport/api/v1/editor.go

```diff
@@ -48,22 +36,84 @@ func (h *Handler) Register(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	statusCode, err := util.PostSupabaseOTPRequest(payload.Email)
-	if statusCode != 0 && err != nil {
+	requestOTP(w, payload.Email)
+
+	return
+}
+
+func requestOTP(w http.ResponseWriter, email string) {
+	statusCode, err := util.PostSupabaseOTPRequest(email)
+	if statusCode != 200 && err != nil {
 		util.WriteResponse(w, statusCode, "Error sending request to supabase: "+err.Error())
 		return
 	}
 	if err != nil {
-		util.WriteResponse(w, http.StatusBadRequest, "Error sending request to supabase: "+err.Error())
+		util.WriteResponse(w, http.StatusInternalServerError, "Error sending request to supabase: "+err.Error())
 		return
 	}
-	if statusCode != 0 && err == nil {
+	if statusCode != 200 {
 		util.WriteResponse(w, http.StatusInternalServerError, fmt.Sprintf("Something unexpected happened: error is nil but statusCode is %d", statusCode))
 		return
 	}
 	return
 }
 
+func (h *Handler) Verify(w http.ResponseWriter, r *http.Request) {
+	data := r.URL.Query().Get("data")
+	if data == "" {
+		util.WriteResponse(w, http.StatusBadRequest, "No data provided")
+		return
+	}
+
+	params, err := parseData(data)
+	if err != nil {
+		util.WriteResponse(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	// Extract email and sub
+	email := params["email"]
```

Comment:
Tady bych vytvořil spíš nějaký struct, kde jsou u fieldů `validate:"required"` (u mailu může být navíc `validate:"required,email"` a bude to kontrolovat formát mailu) pro ty parametry a pak použil validator.New().Struct(&názevTohoStructu) pro validaci fieldů. To může být taky zahrnuto už přímo v té funkci pareseData(data).